File: DSA/PriyankaSingh_590016827/day8question1.c
#include <stdio.h>
#include <ctype.h> // For toupper() and tolower()
#include <string.h> // For strlen()

int main() 
{
    char str[100];
    
    // Input the string 
    printf("Enter a string: ");
    fgets(str, sizeof(str), stdin);
    
    // Remove newline character from fgets
    str[strcspn(str, "\n")] = '\0';
    
    // Calculating string length
    int length = strlen(str);
    
    // Creating copies for case conversion
    char upper[100], lower[100];
    strcpy(upper, str);
    strcpy(lower, str);
    
    // Converting to uppercase and lowercase
    for(int i = 0; i < length; i++) 
    {
        upper[i] = toupper(upper[i]);
        lower[i] = tolower(lower[i]);
    }
    
    printf("\nOriginal string: %s\n", str);
    printf("String length: %d\n", length);
    printf("Uppercase: %s\n", upper);
    printf("Lowercase: %s\n", lower);
    
    return 0;
}

File: DSA/PranyaGupta_500126032/Day8Question1.py
class Solution:
    def StringBasics(self, s):
        length = len(s)
        uprcase = s.upper()
        lwrcase = s.lower()
        return f"Length: {length}, UpperCase: {uprcase}, LowerCase: {lwrcase}"

def main():
    s = input("Enter the String: ")
    sol = Solution()
    result = sol.StringBasics(s)
    print(result)

if __name__ == "__main__":
    main()
